### ISSUE TITLE
Coding of categorical variables

### ISSUE THREAD

**Issue by Pepijn2001** [TPM034A/Q2_2023#13]
Hi,

I feel like I'm not coding the categorical variables correctly. I get a df with 38 columns. The accuracy for my model is very low (10%) and now I'm trying to make a confusion matrix I notice that it's quite large and unclear due to the large amount of columns. I'm not sure what I'm doing wrong, this is my code for coding the categorical variables:

cat_features = df_MLP[['purpose', 'Region']]
cat_features = pd.get_dummies(cat_features, columns=['purpose', 'Region'])

Y = cat_features

The df_MLP is a merged df containing both categorical variables that were asked. Is this the correct way of doing it?

**Comment by spierenbu** [comment]
Different things could affect your predicting power, and you would need more information to solve that issue. Some possible reasons:

- If the numerical features are not scaled, the MLP will perform poorly.
- You may have a misalignment between the columns in you training set and the columns in your test set (if the third column is column C in the train set and column D in the test set, sk-learn may interpret column D as C).
- You do not scale/one-hot encode the test set with the same scaler as in the train set.
